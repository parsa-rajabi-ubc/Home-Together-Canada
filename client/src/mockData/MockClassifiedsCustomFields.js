/**
 * @Author:     Alex Qin
 * @Created:    2021.02.21
 *
 * @Description: mock data for custom fields.
 *
 */

export const rentalsMockCustomFields = {
    price: 800,
    numBed: 5,
    numBath: 3,
    furnished: true,
    petFriendly: false,
    smokingFriendly: false,
    pictures: "pictures"
}

export const houseServicesMockCustomFields = {
    rateAndFees: "666 per day",
    pictures: "pictures"
}

export const agenciesMockCustomFields = {
    rateAndFees: "555 per day",
    pictures: "pictures"
}

export const eventsMockCustomFields = {
    rateAndFees: "666 per day",
    eventDateTime: "2020/2/21 11:00 A.M.",
    contactName: "Alex",
    contactNumber:"778-555-5555",
    pictures: "pictures"
}