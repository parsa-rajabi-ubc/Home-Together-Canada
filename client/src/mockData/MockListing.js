/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.02.20
 *
 * @Description: Mock Listing
 *
 */
import {
    BUSINESS_SERVICE_CATEGORIES, MEMBER_SERVICE_CATEGORIES
} from "../createListing/constants/serviceListingCategoriesText";
import {BUSINESS_CLASSIFIEDS_CATEGORIES} from "../createListing/constants/classifiedListingCategoriesText";

var faker = require('faker');

export const mockServiceListings = [
    //CO-HOUSING 0
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.CO_HOUSING,
        fullDescription: faker.lorem.paragraph(),
        contactName: faker.name.findName(),
        unitsForSale: faker.random.number(),
        unitsForRent: faker.random.number(),
    },
    // SHARED HOME 1
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.SHARED_HOME_SERVICES,
        fullDescription: faker.lorem.paragraph(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // SHARED BUSINESS 2
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.SHARED_BUSINESS_SERVICES,
        fullDescription: faker.lorem.paragraph(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // GOVERNMENT 3
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.GOVERNMENT_SERVICES,
        fullDescription: faker.lorem.paragraph(),
        contactName: faker.name.findName(),
        contactPhoneNumber: faker.phone.phoneNumber(),
    },
    // MEMBER HOME TO SHARE 4
    {
        title: faker.lorem.words(),
        category: MEMBER_SERVICE_CATEGORIES.MEMBER_HOME,
        fullDescription: faker.lorem.paragraph(),

        generalLocation: faker.address.zipCode(),
        monthlyCost: faker.random.number(),
        pictures: faker.image.cats(),
        numBed: faker.random.number(),
        numBath: faker.random.number(),
        petFriendly: faker.random.boolean(),
        smokeFriendly: faker.random.boolean(),
        utilIncluded: faker.random.boolean(),
    }
]

export const mockClassifiedsListings = [
    //RENTALS 0
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.RENTALS,
        fullDescription: faker.lorem.paragraph(),

        price: faker.random.number(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
        numBed: faker.random.number(),
        numBath: faker.random.number(),
        furnished: faker.random.boolean(),
        petFriendly: faker.random.boolean(),
        smokeFriendly: faker.random.boolean(),
    },
    // HOUSE_YARD 1
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.HOUSE_YARD,
        fullDescription: faker.lorem.paragraph(),

        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // LEGAL_SALES 2
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.LEGAL_SALES,
        fullDescription: faker.lorem.paragraph(),

        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // CLASSES_CLUBS 3
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.CLASSES_CLUBS,
        fullDescription: faker.lorem.paragraph(),

        contactName: faker.name.findName(),
        contactPhoneNumber: faker.phone.phoneNumber(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
        eventDateTime: faker.date.soon() + " " + faker.time.recent()
    }
]