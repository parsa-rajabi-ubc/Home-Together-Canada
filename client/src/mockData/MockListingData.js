/**
 * @Author:     Jeff Hatton
 * @Created:    2020.02.16
 *
 * @Description: Const array of mock listings - includes listings for members and businesses
 *
 */

export const mockMemberListings = [
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered",
        monthlyCost: 750,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-14T02:42:39.005Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Unfurnished Room, Utilities/internet Split",
        monthlyCost: 500,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        title: "Basement suite",
        shortDescription: "1 bed 1 bath, nice part of town, parking spot",
        monthlyCost: 1100,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-14T16:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered",
        monthlyCost: 850,
        petFriendly: false,
        smokeFriendly: true,
        datePosted: "2021-02-16T23:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered, Private Kitchenette",
        monthlyCost: 950,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-16T03:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Unfurnished Room, spacious living area ",
        monthlyCost: 775,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-15T11:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Unfurnished Room, large kitchen storage",
        monthlyCost: 750,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-13T18:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, significant storage space",
        monthlyCost: 650,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-13T05:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, private bathroom",
        monthlyCost: 750,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, fast internet",
        monthlyCost: 850,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-16T01:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished, private bath",
        monthlyCost: 950,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-16T02:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered",
        monthlyCost: 750,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-14T08:42:39.000Z"
    },
    {
        title: "Rooms for Rent",
        shortDescription: "3 Furnished Rooms, for students preferably",
        monthlyCost: 650,
        petFriendly: false,
        smokeFriendly: true,
        datePosted: "2021-02-14T07:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, beautiful house",
        monthlyCost: 800,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        title: "Basement suite for Couple",
        shortDescription: "Furnished Large Room, Utilities Covered",
        monthlyCost: 1300,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-13T12:42:39.000Z"
    }
];

export const mockBusinessListings = [
    {
        logo: "public/uploads/logo1",
        title: "Apartments for 2 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-14T22:42:39.000Z"
    },
    {
        logo: "public/uploads/logo2",
        title: "Apartments for 3 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "3 bed 2 bath - furnished",
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo3",
        title: "House with 4 rooms available",
        businessName: "Kelowna House Rentals",
        shortDescription: "4 bed 3 bath - furnished",
        datePosted: "2021-02-14T16:42:39.000Z"
    },
    {
        logo: "public/uploads/logo4",
        title: "Beautiful House On Lakefront For Rent",
        businessName: "Lakefront Properties",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-16T23:42:39.000Z"
    },
    {
        logo: "public/uploads/logo5",
        title: "Vacuum Rentals",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-16T03:42:39.000Z"
    },
    {
        logo: "public/uploads/logo6",
        title: "Lawn Management Services",
        businessName: "Lawns-R-Us",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-15T11:42:39.000Z"
    },
    {
        logo: "public/uploads/logo7",
        title: "Carpet Cleaning Rentals",
        businessName: "Carpet Cleaners",
        shortDescription: "We offer services, or tool rentals",
        datePosted: "2021-02-13T18:42:39.000Z"
    },
    {
        logo: "public/uploads/logo8",
        title: "Plumbing Services Available",
        businessName: "Kelowna Plumbing Inc.",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-13T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo9",
        title: "Apartment Rooms Available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "Apartments of various sizes",
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo10",
        title: "Window Cleaning Services",
        businessName: "Kelowna Glass Cleaners",
        shortDescription: "We clean residential or high rise complexes",
        datePosted: "2021-02-16T01:42:39.000Z"
    },
    {
        logo: "public/uploads/logo11",
        title: "Apartments for 2 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 1 bath - unfurnished",
        datePosted: "2021-02-16T02:42:39.000Z"
    },
    {
        logo: "public/uploads/logo12",
        title: "Apartments for 2 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 2 bath - Unfurnished",
        datePosted: "2021-02-14T08:42:39.000Z"
    },
    {
        logo: "public/uploads/logo13",
        title: "Apartments for 3 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "3 bed 2 bath - Unfurnished",
        datePosted: "2021-02-14T07:42:39.000Z"
    },
    {
        logo: "public/uploads/logo14",
        title: "Senior services",
        businessName: "Senior services",
        shortDescription: "Mobility tools, awareness of area services, financial aid",
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo15",
        title: "Shared Vehicle Insurance",
        businessName: "Kelowna Vehicle Insurance",
        shortDescription: "Insurance and help available for groups sharing cars",
        datePosted: "2021-02-13T12:42:39.000Z"
    }
];