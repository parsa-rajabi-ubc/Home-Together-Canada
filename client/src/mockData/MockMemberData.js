/**
 * @Author:     Parsa Rajabi
 * @Created:    June 2022
 *
 * @Description: Mock Member data for automated testing
 *
 */
const {faker} = require("@faker-js/faker");

const MOCK_MEMBER_DATA = {
    firstName: faker.name.firstName(),
    lastName: faker.name.lastName(),
    email: 'tech@hometogether.ca',
    phoneNumber: {
        areaCode: faker.random.numeric(3),
        prefix: faker.random.numeric(3),
        suffix: faker.random.numeric(4)
    },
    address: {
        street: faker.address.streetAddress(),
        apt: faker.address.buildingNumber(),
        city: faker.address.city(),
        province: "Alberta",
        postCode: "T3T1T3"
    },
    monthlyRent: {
        min: Math.floor(Math.random() * 999) + 500,
        max: Math.floor(Math.random() * 2500) + 1000,
    },
    preferences:{
        petFriendly: faker.datatype.boolean(),
        smoking: faker.datatype.boolean(),
        mobile: faker.datatype.boolean(),
        allergies: faker.datatype.boolean(),
        religion: faker.datatype.boolean(),
        diet: faker.datatype.boolean(),
        hasHome: faker.datatype.boolean(),
        interestInBuyingHome: faker.datatype.boolean(),
    },
    aboutSelf: faker.random.words(2),

    sharingPreferences:{
        genders: ["Male", "Female", "Other"],
        age:{
            min: Math.floor(Math.random() * 29) + 18,
            max: Math.floor(Math.random() * 60) + 30,
        },
        overlapBudget:{
            min: Math.floor(Math.random() * 999) + 500,
            max: Math.floor(Math.random() * 2500) + 1000,
        }
    },
    accountDetails:{
        username: "TestAccount"+faker.random.numeric(2),
        password: "Bose1234"
    }

}

module.exports = {MOCK_MEMBER_DATA}