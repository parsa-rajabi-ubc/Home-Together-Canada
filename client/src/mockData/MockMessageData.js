/**
 * @Author:     Jeff Hatton
 * @Created:    2020.03.03
 *
 * @Description: Const array of mock messages - includes multiple conversations, most between messageMember and other members with differing timestamps
 *
 */

export const mockMessages = [
    {
        id: 1,
        messageContent: "message content 1 all on different days and times",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-12T02:42:39.005Z"
    },
    {
        id: 2,
        messageContent: "message content 2",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-01T02:42:39.005Z"
    },
    {
        id: 3,
        messageContent: "message content 3",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-02-14T08:42:39.005Z"
    },
    {
        id: 4,
        messageContent: "message content 4",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-12T02:42:39.005Z"
    },
    {
        id: 5,
        messageContent: "message content 5",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2020-03-13T02:42:39.075Z"
    },
    {
        id: 6,
        messageContent: "message content 6",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-02-17T02:43:39.005Z"
    },
    {
        id: 7,
        messageContent: "message content 7",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2021-01-19T09:42:39.005Z"
    },
    {
        id: 8,
        messageContent: "message content 8",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2021-01-11T02:42:39.005Z"
    },
    {
        id: 9,
        messageContent: "message content 9 has timestamp before a smaller message id",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-10T02:42:39.005Z"
    },
    {
        id: 10,
        messageContent: "message content 10",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2021-02-10T02:42:39.005Z"
    }
];