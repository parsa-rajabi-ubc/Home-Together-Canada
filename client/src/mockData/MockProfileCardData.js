/**
 * @Author:     Jeff Hatton
 * @Created:    2020.01.12
 *
 * @Description: Const array of mock profiles
 *
 */

import React from 'react';

const mockProfiles = [
    {
        username: "AliceinWonderLand",
        gender: "Female",
        age: 13,
        status: "Single",
        minRent: 200,
        maxRent: 550,
        pet: false,
        smoke: false,
        religion: false,
        diet: false,
    },
    {
        username: "Babar",
        gender: "Male",
        age: 30,
        status: "Couple With Children",
        minRent: 2000,
        maxRent: 5500,
        pet: true,
        smoke: true,
        religion: true,
        diet: true,
    },
    {
        username: "CREATINE",
        gender: "Other",
        age: 20,
        status: "Single",
        minRent: 800,
        maxRent: 1200,
        pet: false,
        smoke: true,
        religion: true,
        diet: true,
    },
    {
        username: "DonaldDuck",
        gender: "Male",
        age: 35,
        status: "Single Parent",
        minRent: 500,
        maxRent: 600,
        pet: false,
        smoke: false,
        religion: true,
        diet: true,
    },
    {
        username: "EusticeOfNarnia",
        gender: "Male",
        age: 70,
        status: "Single",
        minRent: 1300,
        maxRent: 2300,
        pet: false,
        smoke: false,
        religion: false,
        diet: true,
    },
    {
        username: "Foxtrot",
        gender: "Other",
        age: 65,
        status: "Couple",
        minRent: 1400,
        maxRent: 1600,
        pet: false,
        smoke: false,
        religion: true,
        diet: false,
    },
    {
        username: "Grinch",
        gender: "Other",
        age: 85,
        status: "Single",
        minRent: 2500,
        maxRent: 3700,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "Hydra",
        gender: "Female",
        age: 45,
        status: "Couple With Children",
        minRent: 1200,
        maxRent: 2550,
        pet: true,
        smoke: false,
        religion: false,
        diet: false,
    },
    {
        username: "InnigoMontoya",
        gender: "Male",
        age: 50,
        status: "Single Parent",
        minRent: 300,
        maxRent: 575,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "JamesBond",
        gender: "Male",
        age: 70,
        status: "Single",
        minRent: 200,
        maxRent: 550,
        pet: false,
        smoke: true,
        religion: false,
        diet: true,
    },
    {
        username: "Kirby",
        gender: "Other",
        age: 44,
        status: "Existing Group",
        minRent: 200,
        maxRent: 2550,
        pet: false,
        smoke: false,
        religion: false,
        diet: false,
    },
    {
        username: "Link",
        gender: "Male",
        age: 20,
        status: "Couple",
        minRent: 500,
        maxRent: 700,
        pet: true,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "Mario",
        gender: "Male",
        age: 70,
        status: "Single",
        minRent: 900,
        maxRent: 2600,
        pet: false,
        smoke: true,
        religion: true,
        diet: false,
    },
    {
        username: "Nero",
        gender: "Male",
        age: 53,
        status: "Existing Group",
        minRent: 3300,
        maxRent: 6650,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "OddDuck",
        gender: "Female",
        age: 23,
        status: "Single",
        minRent: 400,
        maxRent: 1950,
        pet: false,
        smoke: true,
        religion: true,
        diet: true,
    },
    {
        username: "PlatoInPerson",
        gender: "Male",
        age: 70,
        status: "Single Parent",
        minRent: 100,
        maxRent: 650,
        pet: true,
        smoke: false,
        religion: false,
        diet: true,
    },
    {
        username: "QWERTYworker",
        gender: "Male",
        age: 47,
        status: "Single",
        minRent: 2200,
        maxRent: 3550,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "Rattata",
        gender: "Female",
        age: 34,
        status: "Single",
        minRent: 700,
        maxRent: 1250,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "Sierra",
        gender: "Female",
        age: 85,
        status: "Single",
        minRent: 800,
        maxRent: 1150,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "TooFriendly",
        gender: "Male",
        age: 51,
        status: "Single",
        minRent: 700,
        maxRent: 1050,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "UltimateRoomie123",
        gender: "Other",
        age: 25,
        status: "Single",
        minRent: 2400,
        maxRent: 2550,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "Ventriloquist45",
        gender: "Male",
        age: 63,
        status: "Existing Group",
        minRent: 900,
        maxRent: 2050,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "Wario",
        gender: "Male",
        age: 32,
        status: "Single",
        minRent: 700,
        maxRent: 1350,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "XenaThePrincess",
        gender: "Female",
        age: 40,
        status: "Single",
        minRent: 1100,
        maxRent: 3550
    },
    {
        username: "YogurtFanatic",
        gender: "Male",
        age: 32,
        status: "Couple With Children",
        minRent: 1450,
        maxRent: 1550,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    },
    {
        username: "Zelda995",
        gender: "Female",
        age: 28,
        status: "Couple",
        minRent: 700,
        maxRent: 980,
        pet: false,
        smoke: true,
        religion: false,
        diet: false,
    }
]

export default mockProfiles;
