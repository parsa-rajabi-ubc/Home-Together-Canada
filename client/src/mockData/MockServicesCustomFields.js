/**
 * @Author:     Alex Qin
 * @Created:    2021.02.18
 *
 * @Description: mock data for custom fields.
 *
 */


export const memberMockCustomFields = {
    streetLine1: "123 Candy Lane",
    streetLine2: "",
    city: "Kelowna",
    province: "BC",
    postalCode: "V1X 1X3",
    homeShareMonthlyCost: 800,
    numBed: 5,
    numBath: 3,
    utilIncluded: true,
    petFriendly: false,
    smokeFriendly: false,
    pictures: ['uploads/listings/1/testimage1.png', 'uploads/listings/1/testimage2.png']
}

export const cohousingMockCustomFields = {
    contactName: "Alex",
    unitsForSale: 5,
    unitsForRent: 5,
}

export const homeServicesMockCustomFields = {
    rateAndFees: "666 per day",
    pictures: ['uploads/listings/1/testimage1.png', 'uploads/listings/1/testimage2.png']
}

export const governmentMockCustomFields = {
    contactPerson: "Alex",
    phoneNumber: '123-123-1234'
}