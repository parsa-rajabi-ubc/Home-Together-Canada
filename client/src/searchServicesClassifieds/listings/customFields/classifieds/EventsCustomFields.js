/**
 * @Author:     Alex Qin
 * @Created:    2021.02.21
 *
 * @Description: Re-usable Events custom fields.
 *
 */
import PropTypes from "prop-types";
import React from 'react';
import HTC_Logo from "../../../../images/HTC_Logo.jpg";

const EVENTS_CUSTOM_FIELDS_TEXT = {
    RATE_AND_FEES: "Rates and fees (CAD)",
    CONTACT_NAME: "Contact Name",
    CONTACT_NUMBER: "Contact Phone Number",
    EVENTS_DATE_AND_TIME: "Event dates and times",
    PICTURES: "Pictures"
}

const EventsCustomFields = (props) => {
    const { rateAndFees, contactName, contactNumber, eventDateTime, pictures} = props;

    return (
        <div>
            <label className={"label-result"}>{EVENTS_CUSTOM_FIELDS_TEXT.RATE_AND_FEES}</label>
            <p> {rateAndFees}</p>

            <label className={"label-result"}>{EVENTS_CUSTOM_FIELDS_TEXT.CONTACT_NAME}</label>
            <p> {contactName}</p>

            <label className={"label-result"}>{EVENTS_CUSTOM_FIELDS_TEXT.CONTACT_NUMBER}</label>
            <p> {contactNumber}</p>

            <label className={"label-result"}>{EVENTS_CUSTOM_FIELDS_TEXT.EVENTS_DATE_AND_TIME}</label>
            <p> {eventDateTime}</p>

            <label className={"label-result"}>{EVENTS_CUSTOM_FIELDS_TEXT.PICTURES}</label>

            {/*TODO: replace HTC_Logo with picture string from DB*/}
            <img src={HTC_Logo} alt={""}/>
        </div>
    );
}

EventsCustomFields.propTypes = {
    rateAndFees: PropTypes.string.isRequired,
    contactName: PropTypes.string.isRequired,
    contactNumber: PropTypes.string.isRequired,
    eventDateTime: PropTypes.string.isRequired,
    pictures: PropTypes.string.isRequired
}

export default EventsCustomFields;