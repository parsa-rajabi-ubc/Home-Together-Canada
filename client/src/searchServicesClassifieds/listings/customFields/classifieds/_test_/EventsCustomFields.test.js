/**
 * @Author:     Alex Qin
 * @Created:    2021.02.21
 *
 * @Description: EventsCustomFields snapshot test. Returns a snapshot.
 *
 */
import React from 'react';
import renderer from  'react-test-renderer'
import {eventsMockCustomFields} from "../../../../../mockData/MockClassifiedsCustomFields";
import EventsCustomFields from "../EventsCustomFields";

describe('EventsCustomFields', () => {
    it("should render correctly regardless of properties", () => {
        // given
        const props = {
            ...eventsMockCustomFields,
        };
        //when
        const component = renderer.create(<EventsCustomFields {...props}/>);
        const tree = component.toJSON();
        //then
        expect(tree).toMatchSnapshot();
    });
});