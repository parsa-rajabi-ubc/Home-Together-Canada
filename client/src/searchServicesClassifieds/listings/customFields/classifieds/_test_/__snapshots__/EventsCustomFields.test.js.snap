// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsCustomFields should render correctly regardless of properties 1`] = `
<div>
  <label
    className="label-result"
  >
    Rates and fees (CAD)
  </label>
  <p>
     
    666 per day
  </p>
  <label
    className="label-result"
  >
    Contact Name
  </label>
  <p>
     
    Alex
  </p>
  <label
    className="label-result"
  >
    Contact Phone Number
  </label>
  <p>
     
    778-555-5555
  </p>
  <label
    className="label-result"
  >
    Event dates and times
  </label>
  <p>
     
    2020/2/21 11:00 A.M.
  </p>
  <div>
    <label
      className="label-result"
    >
      Pictures
    </label>
    <div>
      <div
        className="lightbox-container"
      >
        <img
          className="lightbox-img-thumbnail"
          onClick={[Function]}
          src="/uploads/listings/1/testimage1.png"
          style={
            Object {
              "height": "80px",
              "width": "80px",
            }
          }
        />
        <img
          className="lightbox-img-thumbnail"
          onClick={[Function]}
          src="/uploads/listings/1/testimage2.png"
          style={
            Object {
              "height": "80px",
              "width": "80px",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
